### PR TITLE
(Release-4.0.0) HDS-2069 Select accessibility fixes

### PR DESCRIPTION
**Pull request by NikoHelle**
## Description

Latest accessibility auditing found these issues fixed in this PR:
- search errors are not announced to screen reader like other filter/search notifications
- single select group labels can be left out of navigation
- "show less" -button should not lose focus on click. Also a screen reader notification should be announced.

Also added:
- clearable -prop, so select cannot be cleared after selection
- themes for tags. Added also same `--computed-` that are in new scss components.
- fixed some translation keys
- hidden selected option counts must be re-calculated with resizeObserver when component is resized.

## Related Issue

Closes [HDS-2069](https://helsinkisolutionoffice.atlassian.net/browse/HDS-2069)

## How Has This Been Tested?

Jest, e2e

## Demos:

Links to demos are in the comments

## Add to changelog

- changelog in [docs PR](https://github.com/City-of-Helsinki/helsinki-design-system/pull/1389)

[HDS-2069]: https://helsinkisolutionoffice.atlassian.net/browse/HDS-2069?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: packages/react/src/components/select/Select.module.scss [hunk 1]
```diff
@@ -252,23 +255,15 @@
...
-        --menu-item-background-hover: var(--color-bus-light);
-        --menu-item-background-selected: var(--color-bus);
-        --menu-item-background-selected-hover: var(--color-bus);
-        --menu-item-color-selected: var(--color-white);
-        --menu-item-color-selected-hover: var(--color-white);
-
-        &.disabledOption.disabledOption {
-          color: var(--color-black-40);
-        }
+        --computed-menu-item-background-hover: var(--menu-item-background-hover, var(--color-bus-light));
```
Comment (mrTuomoK) [inline packages/react/src/components/select/Select.module.scss:262]: hmm... should all these `*-background-*` be `*-background-color-*`? and if there's more of these in which it only affects a -color?

**Review by mrTuomoK — CHANGES_REQUESTED** [review]
Commented about the -background- ->  -background-color-

File: packages/react/src/components/select/Select.module.scss (hunk 1 above)
Comment (NikoHelle) [reply packages/react/src/components/select/Select.module.scss:262]: The naming is copied from the old Select, so haven't really thought about it. The only thing that can change in the background is the color.

Comment (mrTuomoK) [reply packages/react/src/components/select/Select.module.scss:262]: So then I'd prefer it to be *-color* :D everywhere

Comment (NikoHelle) [reply packages/react/src/components/select/Select.module.scss]: Changed!

**Review by mrTuomoK — APPROVED** [review]
Good work again!

**Pull request merged**